### PR TITLE
Slightly simplify

### PR DESCRIPTION
**Pull request by okurz**
* Simplify counter assignment in aggregates
* approver: Extract _handle_http_error function

### CODE REVIEW

**Review by mimi1vx — COMMENTED** [review]
Small stupid question ? Why ..

This isnt perl to have everything on one line ...

amd why extract code to separatred function to use in one place ?

**Comment by okurz** [comment]
> Small stupid question ? Why ..
> 
> This isnt perl to have everything on one line ...

This does not have to do with Perl or Python but is applying the generic concept of conditional value assignment. Less procedural code when the only thing we want is actually a value "counter" applied, just with values depending on conditions

> amd why extract code to separatred function to use in one place ?

to reduce complexity on each level with abstraction. A side-effect is less indendation. Same reason why the Linux kernel coding style only allows two levels of indendation.

**Comment by mimi1vx** [comment]
> > Small stupid question ? Why ..
> > This isnt perl to have everything on one line ...
> 
> This does not have to do with Perl or Python but is applying the generic concept of conditional value assignment. Less procedural code when the only thing we want is actually a value "counter" applied, just with values depending on conditions
> 
> > amd why extract code to separatred function to use in one place ?
> 
> to reduce complexity on each level with abstraction. A side-effect is less indendation. Same reason why the Linux kernel coding style only allows two levels of indendation.

so .. lower readability for better code .., btw kernel has millions of lines so some restrictive rules are really needed

**Review by mimi1vx — APPROVED** [review]

**Comment by grisu48** [comment]
I'm late for the party but also like to add that it's in general not an improvement to make code more compact and less readable. See [the rule of six](https://davidamos.dev/the-rule-of-six/).

Less code that does more typically means increased mental effort to understand what's going on.

File: openqabot/types/aggregate.py [hunk 1]
```diff
@@ -47,18 +47,14 @@ def __repr__(self):
         return f"<Aggregate product: {self.product}>"
 
     @staticmethod
-    def get_buildnr(repohash: str, old_repohash: str, cur_build: str) -> str:
+    def get_buildnr(repohash: str, old_repohash: str, build: str) -> str:
         today = date.today().strftime("%Y%m%d")
         build = today
```
Comment (baierjan) [inline openqabot/types/aggregate.py:52]: Variable name clash here, see #104

File: openqabot/types/aggregate.py [hunk 2]
```diff
@@ -47,18 +47,14 @@ def __repr__(self):
...
+        if build.startswith(today) and repohash == old_repohash:
             raise SameBuildExists
 
-        if cur_build.startswith(today):
-            counter = int(cur_build.split("-")[-1]) + 1
-        else:
-            counter = 1
-
+        counter = int(build.split("-")[-1]) + 1 if build.startswith(today) else 1
         return f"{build}-{counter}"
```
Comment (baierjan) [inline openqabot/types/aggregate.py:58]: and here

**Pull request merged**